src/services/user.services.ts

```diff
@@ -1,23 +1,116 @@
-import { AuthenticationError } from 'apollo-server';
+import { AuthenticationError, ForbiddenError, ApolloError } from 'apollo-server';
 import { User } from '../sequelize/models/user.models';
-import { unHashPassword } from '../utils/user.utils';
+import { Role } from '../sequelize/models/role.models';
+import { Institution } from '../sequelize/models/institution.models';
+import { unHashPassword, hashPassword } from '../utils/user.utils';
 import { IloginData, Iuser } from '../interfaces/user.interfaces';
-
+import Redis from '../config/redis';
+import sendEmail from '../utils/mailer/send.mailer';
 
 class UserService {
   static async loginUser(user: IloginData) {
     const { email, password } = user;
-    const findUser = await this.findUserByEmail(email);
+    const findUser = await this.findUser(email, 'email');
     if (!findUser || !unHashPassword(password, findUser.password)) {
       throw new AuthenticationError('Invalid email or password');
     }
+    if (findUser.verified === false) throw new AuthenticationError('Sorry, you can not log in. Please check your email and verify your account first.');
     return findUser.get() as Iuser;
   }
 
-  static async findUserByEmail(email: string) {
-    const user = await User.findOne({ where: { email } });
+  static async findUser(credential: string | number, option: string) {
+    let user;
```

Comment:
Could we force users to include some sort of option? Because it will return user undefined in case things go wrong.

Reply:
Of course not. This will be handled easily in frontend. I think it will not be a problem.